### DISCUSSION TITLE
How to correctly handle MissingHeartbeat in PullConsumer?

### DISCUSSION THREAD

**Discussion by wyl1232792** [nats-io/nats.rs#1102] [Q&A]
I followed the demo for consuming stream events using pull consumer like:
```
            let mut messages = consumer.messages().await.unwrap();
            while let Some(res) = messages.next().await {
                       // Err(e) => log::error!(......)
            }
```
and some time, I got this MissingHeartbeat Error.
missed idle heartbeat OrderedError { kind: MissingHeartbeat, source: None }

Should I just ignore it, or consider there's any error.

**Comment by wyl1232792** [comment]
I'm using 
nats-server v2.9.17
async-nats 0.31

**Comment by Jarema** [comment]
The `MissingHeartbeat` means that the client did not get heartbeat in time from the server.
It can mean that the consumer is gone, or that it's just a hiccup in networking/processing.
If the idle hearbeat does not repeat, it probably means the latter. If it constantly repeats, it means that is something wrong with the consumer.
